peer.go

```diff
@@ -247,7 +247,7 @@ func newPeer(conn net.Conn, connReq *connmgr.ConnReq, server *server,
 		outgoingCltvRejectDelta: outgoingCltvRejectDelta,
 
 		sendQueue:     make(chan outgoingMsg),
-		outgoingQueue: make(chan outgoingMsg),
+		outgoingQueue: make(chan outgoingMsg, outgoingQueueLen),
```

Comment:
does this address an existing issue? atm the `queue` method blocks until the message is actually being assigned it's final outgoing order, which i think we may want to keep. if there isn't an issue, then maybe remove `outgoingQueueLen` instead?

Reply:
My rationale was that this would give a speedup when asynchronously sending messages which happens in a couple places in the code.  I guess the question is if blocking until order assignment is desirable?